### PR TITLE
{toolchain} GNU 5.1.0-2.25 (REVIEW)

### PR DESCRIPTION
**Pull request by boegel**
useful information for stuff that will be built on top of this: https://lists.fedoraproject.org/pipermail/devel/2015-February/207549.html

### CODE REVIEW

**Comment by boegel** [comment]
WIP: still being tested

@wpoely86: please review?

File: easybuild/easyconfigs/b/binutils/binutils-2.25-GCC-5.1.0-binutils-2.25.eb [hunk 1]
```diff
@@ -0,0 +1,35 @@
...
+
+homepage = 'http://directory.fsf.org/project/binutils/'
+description = "binutils: GNU binary utilities"
+
+toolchain = {'name': 'GCC', 'version': '5.1.0-binutils-%(version)s'}
+
+sources = [SOURCE_TAR_GZ]
+source_urls = [GNU_SOURCE]
+
+builddependencies = [
```
Comment (wpoely86) [inline easybuild/easyconfigs/b/binutils/binutils-2.25-GCC-5.1.0-binutils-2.25.eb]: add a comment why these are build deps.

File: easybuild/easyconfigs/g/GCC/GCC-5.1.0-binutils-2.25.eb [hunk 2]
```diff
@@ -0,0 +1,52 @@
...
+
+mpfr_version = '3.1.2'
+
+source_urls = [
+    'http://ftpmirror.gnu.org/%(namelower)s/%(namelower)s-%(version)s',  # GCC auto-resolving HTTP mirror
+    'http://ftpmirror.gnu.org/gmp',  # idem for GMP
+    'http://ftpmirror.gnu.org/mpfr',  # idem for MPFR
+    'http://www.multiprecision.org/mpc/download',  # MPC official
+    'ftp://gcc.gnu.org/pub/gcc/infrastructure/',  # GCC dependencies
+    'http://gcc.cybermirror.org/infrastructure/',  # HTTP mirror for GCC dependencies
```
Comment (wpoely86) [inline easybuild/easyconfigs/g/GCC/GCC-5.1.0-binutils-2.25.eb]: add `http://isl.gforge.inria.fr/` for an isl mirror?

File: easybuild/easyconfigs/g/GNU/GNU-5.1.0-2.25.eb [hunk 3]
```diff
@@ -0,0 +1,20 @@
...
+
+name = 'GNU'
+gccver = '5.1.0'
+binutilsver = '2.25'
+version = '%s-%s' % (gccver, binutilsver)
+
+homepage = 'http://www.gnu.org/software/'
+description = "Compiler-only toolchain with GCC and binutils."
+
+toolchain = {'name': 'dummy', 'version': 'dummy'}
```
Comment (wpoely86) [inline easybuild/easyconfigs/g/GNU/GNU-5.1.0-2.25.eb]: `dummy:dummy` works for toolchain deps?

File: easybuild/easyconfigs/z/zlib/zlib-1.2.8-GCC-5.1.0-binutils-2.25.eb [hunk 4]
```diff
@@ -0,0 +1,23 @@
...
+description = """zlib is designed to be a free, general-purpose, legally unencumbered -- that is,
+ not covered by any patents -- lossless data-compression library for use on virtually any
+ computer hardware and operating system."""
+
+toolchain = {'name': 'GCC', 'version': '5.1.0-binutils-2.25'}
+
+sources = [SOURCELOWER_TAR_GZ]
+source_urls = [('http://sourceforge.net/projects/libpng/files/zlib/%(version)s', 'download')]
+
+builddependencies = [('binutils', '2.25', '', True)]
```
Comment (wpoely86) [inline easybuild/easyconfigs/z/zlib/zlib-1.2.8-GCC-5.1.0-binutils-2.25.eb]: add a comment about the why here?

File: easybuild/easyconfigs/z/zlib/zlib-1.2.8-GCC-5.1.0-binutils-2.25.eb [hunk 5]
```diff
@@ -0,0 +1,23 @@
...
+
+homepage = 'http://www.zlib.net/'
+description = """zlib is designed to be a free, general-purpose, legally unencumbered -- that is,
+ not covered by any patents -- lossless data-compression library for use on virtually any
+ computer hardware and operating system."""
+
+toolchain = {'name': 'GCC', 'version': '5.1.0-binutils-2.25'}
+
+sources = [SOURCELOWER_TAR_GZ]
+source_urls = [('http://sourceforge.net/projects/libpng/files/zlib/%(version)s', 'download')]
```
Comment (wpoely86) [inline easybuild/easyconfigs/z/zlib/zlib-1.2.8-GCC-5.1.0-binutils-2.25.eb]: add the official url first? `http://zlib.net/zlib-1.2.8.tar.gz`

**Comment by boegel** [comment]
Test report by @boegel
**SUCCESS**
Build succeeded for 4 out of 4 (4 easyconfigs in this PR)
Linux red hat enterprise linux computenode 6.5, Intel(R) Xeon(R) CPU E5-2680 v3 @ 2.50GHz, Python 2.6.6
See https://gist.github.com/371c2b200b75db5b0322 for a full test report.

File: easybuild/easyconfigs/g/GCC/GCC-5.1.0-binutils-2.25.eb (hunk 2 above)
Comment (boegel) [inline easybuild/easyconfigs/g/GCC/GCC-5.1.0-binutils-2.25.eb]: we haven't done that before, so why now?

File: easybuild/easyconfigs/g/GNU/GNU-5.1.0-2.25.eb (hunk 3 above)
Comment (boegel) [inline easybuild/easyconfigs/g/GNU/GNU-5.1.0-2.25.eb]: Yes, special case.

File: easybuild/easyconfigs/z/zlib/zlib-1.2.8-GCC-5.1.0-binutils-2.25.eb (hunk 5 above)
Comment (boegel) [inline easybuild/easyconfigs/z/zlib/zlib-1.2.8-GCC-5.1.0-binutils-2.25.eb]: well, this is an auto-resolving mirror, which is official (it's also mentioned on zlib.net)

that, and I don't want to make this change in all zlib easyconfigs... ;-)

File: easybuild/easyconfigs/g/GCC/GCC-5.1.0-binutils-2.25.eb (hunk 2 above)
Comment (wpoely86) [inline easybuild/easyconfigs/g/GCC/GCC-5.1.0-binutils-2.25.eb]: why not? More mirrors is robuster?

File: easybuild/easyconfigs/z/zlib/zlib-1.2.8-GCC-5.1.0-binutils-2.25.eb (hunk 5 above)
Comment (wpoely86) [inline easybuild/easyconfigs/z/zlib/zlib-1.2.8-GCC-5.1.0-binutils-2.25.eb]: but but but, it's evil sourceforge! They have turned to the dark side!

But I can live with it, for now :wink:

File: easybuild/easyconfigs/g/GCC/GCC-5.1.0-binutils-2.25.eb (hunk 2 above)
Comment (boegel) [inline easybuild/easyconfigs/g/GCC/GCC-5.1.0-binutils-2.25.eb]: Sure, but not in this PR (since we should do it in all GCC easyconfigs).

**Comment by boegel** [comment]
Test report by @boegel
**SUCCESS**
Build succeeded for 11 out of 11 (8 easyconfigs in this PR)
Linux RHEL 6.6, Intel(R) Xeon(R) CPU E5-2670 0 @ 2.60GHz, Python 2.6.6
See https://gist.github.com/4c2004fe990ff1190ba3 for a full test report.

**Comment by boegel** [comment]
Test report by @boegel
**FAILED**
Build succeeded for 4 out of 11 (8 easyconfigs in this PR)
Linux SLES 11_SP2, Intel(R) Xeon(R) CPU E5-2650 v2 @ 2.60GHz, Python 2.6.9
See https://gist.github.com/86a7b805c1baf4080290 for a full test report.

File: easybuild/easyconfigs/z/zlib/zlib-1.2.8-GCC-5.1.0-binutils-2.25.eb (hunk 5 above)
Comment (fgeorgatos) [inline easybuild/easyconfigs/z/zlib/zlib-1.2.8-GCC-5.1.0-binutils-2.25.eb]: upstream evilness can be tamed by using hashes;
IMHO, we could be liberal in source_urls if hashes are defined consistently.. (for future PRs)

**Comment by wpoely86** [comment]
Test report by @wpoely86
**FAILED**
Build succeeded for 2 out of 4 (1 easyconfigs in this PR)
Linux debian 7.8, Intel(R) Xeon(R) CPU            5130  @ 2.00GHz, Python 2.7.3
See https://gist.github.com/f41086d7046236ed279a for a full test report.

**Comment by boegel** [comment]
@wpoely86: `configure: error: C compiler cannot create executables`, any clues in `config.log`?

**Comment by wpoely86** [comment]
Test report by @wpoely86
**FAILED**
Build succeeded for 0 out of 2 (1 easyconfigs in this PR)
Linux debian 7.8, Intel(R) Xeon(R) CPU            5130  @ 2.00GHz, Python 2.7.3
See https://gist.github.com/76610e9caa4d8a94d45d for a full test report.

**Comment by wpoely86** [comment]
yeah, it's my fault. Messed up zlib install.

**Comment by boegel** [comment]
Test report by @boegel
**SUCCESS**
Build succeeded for 9 out of 9 (8 easyconfigs in this PR)
Linux SLES 11_SP2, Intel(R) Xeon(R) CPU E5-2650 v2 @ 2.60GHz, Python 2.6.9
See https://gist.github.com/a5d39c85ae90de2b86c5 for a full test report.

**Comment by wpoely86** [comment]
Test report by @wpoely86
**SUCCESS**
Build succeeded for 6 out of 6 (1 easyconfigs in this PR)
Linux debian 7.8, Intel(R) Xeon(R) CPU            5130  @ 2.00GHz, Python 2.7.3
See https://gist.github.com/e70f30694318efbbaf66 for a full test report.

**Comment by boegel** [comment]
OK, this is going in, successful test reports on RHEL, Debian *and* SLES.

Thanks @wpoely86 for reviewing/testing!

**Pull request merged**